codespell_lib/tests/test_dictionary.py

```diff
@@ -116,8 +113,8 @@ def _check_err_rep(
     fname: str,
     languages: Tuple[Iterable[str], Iterable[str]],
 ) -> None:
-    assert whitespace.search(err) is None, "error %r has whitespace" % err
-    assert "," not in err, "error %r has a comma" % err
+    assert whitespace.search(err) is None, f"error {err!r} has whitespace"
+    assert "," not in err, f"error {err!r} has a comma"
```

Comment:
Why explicitly specify `!r` with f-strings but not with the `%` formatting operator?